sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/query/CosmosQuery.java

```diff
@@ -72,17 +74,52 @@ public int getLimit() {
         return limit;
     }
 
+    /**
+     * To get offset number
+     *
+     * @return long offset
+     */
+    public long getOffset() {
+        return offset;
+    }
+
     /**
      * To set limit number
      *
      * @param limit int
+     * @deprecated use withLimit instead
      */
+    @Deprecated
     public void setLimit(int limit) {
         if (this.limit == 0) {
             this.limit = limit;
         }
     }
 
+    /**
+     * To set limit number
+     *
+     * @param limit int
+     */
+    public void withLimit(int limit) {
+        if (this.limit == 0) {
+            this.limit = limit;
+        }
+    }
+
+    /**
+     * To set limit number and offset number
+     *
+     * @param offset long
+     * @param limit int
+     */
+    public void withOffsetAndLimit(long offset, int limit) {
+        if (this.limit == 0) {
+            this.limit = limit;
+        }
+        this.offset = offset;
```

Comment:
do we need to do the same checking as limit?  (can only set once?)

Reply:
Done!